File: internal/domain/media.go
package domain

import (
	"fmt"
	"path/filepath"
	"strings"
)

type InputKind string

const (
	InputKindVideo InputKind = "video"
	InputKindImage InputKind = "image"
	InputKindGIF   InputKind = "gif"
)

type MediaInfo struct {
	Width           int
	Height          int
	FPS             float64
	DurationSeconds float64
	HasAudio        bool
	FormatName      string
	CodecName       string
	BitrateBps      int64
	InputSizeBytes  int64
}

var (
	videoExts = map[string]struct{}{".mp4": {}, ".mov": {}, ".webm": {}, ".mkv": {}, ".avi": {}}
	imageExts = map[string]struct{}{".png": {}, ".jpg": {}, ".jpeg": {}, ".webp": {}}
	gifExts   = map[string]struct{}{".gif": {}}
)

func DetectInputKind(path string) (InputKind, error) {
	ext := strings.ToLower(filepath.Ext(path))
	if _, ok := gifExts[ext]; ok {
		return InputKindGIF, nil
	}
	if _, ok := imageExts[ext]; ok {
		return InputKindImage, nil
	}
	if _, ok := videoExts[ext]; ok {
		return InputKindVideo, nil
	}
	return "", fmt.Errorf("unsupported input: %s", path)
}

File: python/core/media.py
from __future__ import annotations

from dataclasses import dataclass
from enum import Enum
from pathlib import Path


class InputKind(str, Enum):
    VIDEO = "video"
    IMAGE = "image"
    GIF = "gif"


@dataclass
class MediaInfo:
    width: int = 0
    height: int = 0
    fps: float = 0.0
    duration_seconds: float = 0.0
    has_audio: bool = False
    format_name: str = ""
    codec_name: str = ""
    bitrate_bps: int = 0
    input_size_bytes: int = 0


VIDEO_EXTS = {".mp4", ".mov", ".webm", ".mkv", ".avi"}
IMAGE_EXTS = {".png", ".jpg", ".jpeg", ".webp"}
GIF_EXTS = {".gif"}


def detect_input_kind(path: str) -> InputKind:
    ext = Path(path).suffix.lower()
    if ext in GIF_EXTS:
        return InputKind.GIF
    if ext in IMAGE_EXTS:
        return InputKind.IMAGE
    if ext in VIDEO_EXTS:
        return InputKind.VIDEO
    raise ValueError(f"unsupported input: {path}")
